src/App.js

```diff
@@ -1,13 +1,101 @@
 import React from 'react';
 
+import TodoList from './components/TodoComponents/TodoList';
+import TodoForm from './components/TodoComponents/TodoForm';
+
+import './components/TodoComponents/Todo.css';
+
 class App extends React.Component {
   // you will need a place to store your state in this component.
   // design `App` to be the parent component of your application.
   // this component is going to take care of state, and any change handlers you need to work with your state
+  constructor() {
+    super();
+    this.state = {
+      name: 'Billy`s List',
+      placeholder: 'Add an item',
+      todos: [
+        {
+          task: 'Organize Garage',
+          id: 1528817077286,
+          completed: false
+        },
+        {
+          task: 'Bake Cookies',
+          id: 1528817084358,
+          completed: false
+        },
+        {
+          task: 'Clean More',
+          id: 1528817011158,
+          completed: false
+        }
+      ],
+      todo: "",
+      completedList: []
+    } 
+  }
+
+  addTodo = (text) => {
+
+    const currentTodos = this.state.todos.slice();
+
+    const newItem = {
+      task: text,
+      id: Date.now(),
+      completed: false
+    }
+
+    if (newItem.task === '') {
+      currentTodos.push();
+    }
+    else {
+      currentTodos.push(newItem);
+    }
+    this.setState({todos: currentTodos});
+
+  }
+
+  markComplete = (todo) => {
+    
+    const htmlElementsArray = document.querySelectorAll('li');
```

Comment:
Directly interacting with the DOM in this way is a React Antipattern.

Instead, make sure that you are taking advantage of your event listeners by passing you event through, so that you can interact with its specific target.

ie.
**On your Todo component:**
`onClick={e => props.checkTodo(e, props.todo)}`

**On App.js**:
`markComplete = (e, todo) => {`
`currentTodo = e.currentTarget`

An even BETTER way is to have the class entirely controlled by state. That is a TRUE React way of handling something like this. Here is a small example: 

`className={props.todo.completed ? "completed" : ""}`

So you update the class by updating the completed property of the respective item in state.